### ISSUE TITLE
Problem with reproducing w4a4 using lwc

### ISSUE THREAD

**Issue by Wotoosh** [Hsu1023/DuQuant#5]
Dear author,when I try to reproduce by running run_lwc.sh with
"python main.py \
    --block_size 128 \
    --max_rotation_step 256 \
    --epochs 20 \
    --wbits 4 \
    --abits 4 \
    --model /mypath/llama-7b-hf \
    --lwc \
    --save_dir ./log/myw4a4_lwc \
    --alpha 0.5 \
    --smooth \
    --lac 0.9 \
    --task piqa",
I only get the results of
" 'results': {'piqa': {'acc': 0.5380848748639826,
                      'acc_norm': 0.5353645266594124,
                      'acc_norm_stderr': 0.011636607860111548,
                      'acc_stderr': 0.011631933367846709}},"
Is there any problem or hyperparameter may cause this problem?Many thanks

**Comment by FelixMessi** [comment]
Hi, thank you for your question!

The parameters appear to be correct, as I have reproduced the experiment and obtained results similar to those reported in the paper. Could you confirm if the PPL results are as expected? If the PPL seems reasonable, I suggest checking the version of the ``lm_eval`` package, as it could impact the results. If you could share more details, we can further investigate the issue together.

**Comment by Wotoosh** [comment]
Dear sir, I rerun the script again and tested the perplexity，I got “ INFO wikitext2 : 93.02549743652344. But I have run the script of "run.sh" before and get the similar results of the paper, so I think the lm_eval package is fine. By the way ,when I try to execute the code in the very beginning, it throw errors of"len(optimizer_state["found_inf_per_device"]) > 0AssertionError: No inf checks were recorded for this optimize", after searching on the Internet, I set the line 33 of the uitils.py to"`self._scaler = torch.cuda.amp.GradScaler(enabled=False)`", and add "`loss.requires_grad_(True)`" to the beginning of the __call__ of NativeScalerWithGradNormCount. Was it cause the problem? If it is, how should I properly deal with it. Thank you very much.
